mtls-southbound/docs/cloudshell-tutorial.md

```diff
@@ -0,0 +1,174 @@
+
+```sh
+# create cert.pem certificate and set to allow our VM_IP as Alt.
+openssl req -subj '/CN=ssl.test.local' -x509 -new -newkey rsa:4096 -keyout key.pem -out cert.pem -sha256 -days 365 -nodes -addext "keyUsage = digitalSignature,keyAgreement" -addext "extendedKeyUsage = serverAuth, clientAuth" -addext "subjectAltName = DNS:ssl.test.local, DNS:localhost, IP:127.0.0.1, IP:$VM_IP"
+
+# create PCKS12 export, use 'test' as password
+openssl pkcs12 -export -out client.p12 -inkey key.pem -in cert.pem
+
+# verify certificate, should return OK
+openssl verify -CAfile cert.pem cert.pem
+```
+
+### Set VM cert and nginx config
+
+Now we will sync the cert and key to the VM, and set the nginx config file as well.
+
+```sh
+# copy cert, key and nginx.conf files to VM /etc/nginx dir
+gcloud compute scp cert.pem key.pem nginx.conf $VM_NAME:/etc/nginx --zone=$ZONE --project $PROJECT_ID
+
+# restart nginx to apply config
+gcloud compute ssh $VM_NAME --zone=$ZONE --project=$PROJECT_ID --command="sudo nginx -s reload"
+```
+
+### Test calling VM directly with cert and key
+
+Test calling the VM to see if nginx is reachable, and we can successfully call the endpoint with our certificate and key.
+
+```sh
+# call with just the cacert, should get the message "blocked access to mTLS-protected resource"
+curl -v https://$VM_IP --cacert cert.pem
+
+# now call with cert and key, should get the message "access to mTLS-protected resource"
+curl -v https://$VM_IP --cacert cert.pem --key key.pem --cert cert.pem
+```
+
+## Create Apigee Keystore and Proxy
+
+```sh
+# create Apigee keystore
+apigeecli keystores create -n mtls-keystore1 -e $APIGEE_ENV -o $PROJECT_ID -t $(gcloud auth print-access-token)
+
+# create Apigee key and cert
+apigeecli keyaliases create -s test-key1 -f keycertfile -k mtls-keystore1 --key-filepath key.pem --cert-filepath cert.pem -e $APIGEE_ENV -o $PROJECT_ID -t $(gcloud auth print-access-token)
+
+# create Apigee cert
+apigeecli keyaliases create -s test-cert1 -f keycertfile -k mtls-keystore1 --cert-filepath cert.pem -e $APIGEE_ENV -o $PROJECT_ID -t $(gcloud auth print-access-token)
+
+# create Apigee target server
+apigeecli targetservers create -c true -s "$VM_IP" -i true --keyalias test-key1 --keystore mtls-keystore1 -n mtls-service -p 443 --tls true --tlsenforce false --truststore mtls-keystore1 -e $APIGEE_ENV -o $PROJECT_ID -t $(gcloud auth print-access-token)
+
+# deploy Apigee proxy
+apigeecli apis create bundle -f apiproxy --name SecureBackendProxy-v1 -o $PROJECT_ID -e $APIGEE_ENV --ovr -t $(gcloud auth print-access-token)
+```
+
+### Test Apigee API proxy to reach mTLS service
+
+Let's do a test call to the Apigee API proxy to see if we can reach our mTLS backend VM.
+
+```sh
+# get Apigee hostname, this will get the first one from the first envgroup, if not correct then adjust..
+HOSTNAME=$(apigeecli envgroups list -o $PROJECT_ID | jq --raw-output '.environmentGroups[0].hostnames[0]')
```

Comment:
same comment as the README? Can we just create a variable called HOSTNAME as part of the .env file

Reply:
As mentioned I prefer this way since it is elegant and simple, and might be useful to people in other contexts... Why not let little extra cool tricks be used in samples, if they are simple and elegant?  HOSTNAME is a common thing that is always needed in these samples, and this is a nice way to get it...